cpp/include/cudf/strings/string_view.cuh

```diff
@@ -373,24 +373,23 @@ __device__ inline size_type string_view::find_impl(char const* str,
                                                    size_type pos,
                                                    size_type count) const
 {
-  auto const nchars = length();
-  if (!str || pos < 0 || pos > nchars) return npos;
-  if (count < 0) count = nchars;
+  if (!str || pos < 0) { return npos; }
+  if (pos > 0 && pos > length()) { return npos; }
```

Comment:
same as above

Reply:
So this is one of the key optimizations. We are avoiding calling `length()` here which can be expensive.
The `length()` is the number of characters which is calculated by counting (and cached).
The `pos` values is in unit of characters and not bytes but when `pos==0` we don't need to make the call to `length()`.

This case is much more likely since we could be given a column of strings of various sizes and a single `pos` value that is past the end of some of them.